interface/usergroup/addrbook_edit.php

```diff
@@ -105,6 +142,266 @@ function typeSelect(a) {
    $(".specialtyRow").show();
   }
  }
+
+    let lastSearchParams = {};
+    let allResults = []; // Store all fetched results
+    let displayOffset = 0; // Track what we've shown to user
+    let totalResultCount = 0;
+    let limit = 200; // max available from NPPES
+
+    function lookupNPI(loadMore = false) {
+    const npi = $('#form_npi').val().trim();
+    const firstName = $('#form_fname').val().trim();
+    const lastName = $('#form_lname').val().trim();
+    const organization = $('#form_organization').val().trim();
+
+    if (!npi && !lastName && !organization) {
+        alert(xlj('Please enter search criteria'));
+        return;
+    }
+
+    // Reset if new search
+    if (!loadMore) {
+        currentSkip = 0;
+        displayOffset = 0;
+        allResults = [];
+        lastSearchParams = { npi, firstName, lastName, organization };
+    }
+
+    // Check if we need to fetch more from API or just display cached results
+    if (displayOffset < allResults.length) {
+        // We have cached results, just display next
+        displayNextBatch();
+        return;
+    }
+
+    // Need to fetch more
+    let queryParams = [];
+    queryParams.push('csrf_token=' + encodeURIComponent(<?php echo js_escape(CsrfUtils::collectCsrfToken()); ?>));
+    if (npi) queryParams.push('number=' + encodeURIComponent(npi));
+    if (firstName) queryParams.push('first_name=' + encodeURIComponent(firstName + '*'));
+    if (lastName) queryParams.push('last_name=' + encodeURIComponent(lastName + '*'));
+    if (organization) {
+        queryParams.push('organization_name=' + encodeURIComponent(organization + '*'));
+        queryParams.push('enumeration_type=NPI-2'); // Filter for organizations only
+    }
+    queryParams.push('limit=' + limit);
+    queryParams.push('skip=' + currentSkip);
+
+    const proxyUrl = 'npi_lookup.php?' + queryParams.join('&');
+    const resultsDiv = $('#npi-lookup-results');
+
+    if (loadMore) {
+        resultsDiv.append('<div class="npi-loading"><i class="fa fa-spinner fa-spin"></i>' + <?php echo xlj("Loading more ..."); ?> + '</div>');
+    } else {
+        resultsDiv.show().html('<div class="npi-loading"><i class="fa fa-spinner fa-spin"></i>' + <?php echo xlj("Searching..."); ?> + '</div>');
```

Comment:
I am not sure that this will fully escape since getting placed into the html. Would do:
```
resultsDiv.show().html('<div class="npi-loading"><i class="fa fa-spinner fa-spin"></i>' + jsText(xl("Searching...")) + '</div>');
```